index.js

```diff
@@ -96,14 +96,17 @@ async function rateLimitPlugin (fastify, settings) {
   const run = Symbol('rate-limit-did-run')
   pluginComponent.run = run
   fastify.decorateRequest(run, false)
-
   if (!fastify.hasDecorator('rateLimit')) {
     // The rate limit plugin can be registered multiple times but decorate throws if called multiple times for the same field
     fastify.decorate('rateLimit', function rateLimit (options) {
       let params = globalParams
       if (options) {
         params = makeParams(options)
       }
+      if (params.timeWindow && params.timeWindow !== pluginComponent.store.timeWindow) {
+        const child = pluginComponent.store.child(Object.assign({}, { routeInfo: {} }, params))
```

Comment:
Need to clone `pluginComponent` before replacing the property. Otherwise, it is modifying the global one.